external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -340,9 +340,19 @@ private[kafka010] class KafkaSourceProvider extends DataSourceRegister
     // Validate user-specified Kafka options
 
     if (caseInsensitiveParams.contains(s"kafka.${ConsumerConfig.GROUP_ID_CONFIG}")) {
-      throw new IllegalArgumentException(
-        s"Kafka option '${ConsumerConfig.GROUP_ID_CONFIG}' is not supported as " +
-          s"user-specified consumer groups are not used to track offsets.")
+      logWarning(
+        s"It is not recommended to set Kafka option 'kafka.${ConsumerConfig.GROUP_ID_CONFIG}'. " +
```

Comment:
The long string for log message is duplicated 3 times. Could we move this and reuse it like `KafkaSourceProvider.INSTRUCTION_FOR_FAIL_ON_DATA_LOSS_FALSE` and `KafkaSourceProvider.INSTRUCTION_FOR_FAIL_ON_DATA_LOSS_TRUE`?

Reply:
Good point. Updated.